config/application.rb

```diff
@@ -35,6 +35,11 @@ class Application < Rails::Application
     # Only loads a smaller set of middleware suitable for API only apps.
     # Middleware like session, flash, cookies can be added back manually.
     # Skip views, helpers and assets when generating a new resource.
-    config.api_only = true
+    config.middleware.insert_before 0, Rack::Cors do
+      allow do
+         origins 'http://localhost:3000'
```

Comment:
This is environment specific ie the origins will differ on your development environment vs on your production environment etc. 
Can you either use credentials.yml or use the environment files and make the origin's value environment specific?

I was trying to find if there is any recommend way by the gem itself, but could not. I found this stackoverflow answer https://stackoverflow.com/a/49985771/2440312 and either of the approach used in here looks good to me.